pandas/io/formats/style.py

```diff
@@ -1023,7 +1023,8 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             )
 
         for cn in attrs.columns:
-            for rn, c in attrs[[cn]].itertuples():
+            ser = attrs[cn]
```

Comment:
```suggestion
            j = self.columns.get_loc(cn)
            ser = attrs[cn]
```
(remove repeated j lookup below)

Reply:
can actually go further and just do `for j, cn in enumerate(attrs.columns):`, but will ask you to do that in #43287

Reply:
can't becuase of `subset`, we are cross matching objects:
i.e. 
```
self.columns ~ ["a", "b", "c", "d"]
attrs.columns ~ ["b", "d"]
=> j ~ [1, 3]  # not enumerate
```